### PR TITLE
Changed DataTable to add theme dataTable.icons.sortable

### PR DESCRIPTION
**Pull request by ericsoderberghp**
#### What does this PR do?

Changed DataTable to add theme dataTable.icons.sortable

#### Where should the reviewer start?

Header.js

#### What testing has been done on this PR?

Added story

#### How should this be manually tested?

story

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/4446

#### Do the grommet docs need to be updated?

automatic

#### Should this PR be mentioned in the release notes?

yes

#### Is this change backwards compatible or is it a breaking change?

backwards compatible

### CODE REVIEW

File: src/js/components/DataTable/Header.js [hunk 1]
```diff
@@ -66,14 +66,17 @@ const Header = ({
...
-              const Icon =
-                onSort &&
-                sortable !== false &&
-                sort &&
-                sort.property === property &&
-                theme.dataTable.icons[
-                  sort.direction !== 'asc' ? 'ascending' : 'descending'
-                ];
+              let Icon;
+              if (onSort && sortable !== false) {
```
Comment (ShimiSun) [inline src/js/components/DataTable/Header.js:70]: I think the icon should still work if `sortable` isn't false and `onSort` wasn't defined.

File: src/js/components/DataTable/doc.js [hunk 2]
```diff
@@ -329,6 +329,11 @@ export const themeDoc = {
     type: 'React.Element',
     defaultValue: '<FormDown />',
   },
+  'dataTable.icons.sortable': {
+    description: 'The icon indicating a column can be sorted.',
```
Comment (ShimiSun) [inline src/js/components/DataTable/doc.js:333]: I'm a little confused about whether the icon will be placed in addition to the up/down icon.
Edit: after looking on storybook, we should add a comment that this is an interim icon until the sort action will take place.

File: src/js/components/DataTable/stories/Custom.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+const customTheme = {
+  global: {
+    font: {
+      family: 'Helvetica',
+    },
+  },
+  dataTable: {
+    icons: {
+      sortable: FormNext,
```
Comment (ShimiSun) [inline src/js/components/DataTable/stories/Custom.js:19]: I personally found this icon to be confusing and too similar to the Next/Prev icons. can we use a different icon that will stand out?

File: src/js/components/DataTable/Header.js (hunk 1 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/Header.js:70]: The point of the icon is to indicate that the user could sort on that column. We don't want to show the icon if the user cannot sort on the column.

File: src/js/components/DataTable/doc.js (hunk 2 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:333]: The icon is intended as an indicator that the column could be sorted. So, it's not exactly "interim". The user might never sort, and that's just fine.

File: src/js/components/DataTable/stories/Custom.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/stories/Custom.js:19]: I scanned through grommet-icons looking for something that might work, to no avail. :(
Perhaps we should add an icon to grommet-icons?

File: src/js/components/DataTable/Header.js (hunk 1 above)
Comment (ShimiSun) [reply src/js/components/DataTable/Header.js:70]: I think my point is that one can use `sortable` prop without `onSort` and still expect the new icon functionality to work as expected. For example, this story https://storybook.grommet.io/?path=/story/datatable--controlled-grouped
uses only `sortable` prop, without `onSort` and I can see a use case of it using the new sortable icon.

File: src/js/components/DataTable/stories/Custom.js (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/DataTable/stories/Custom.js:19]: https://github.com/grommet/grommet-icons/issues/207

File: src/js/components/DataTable/doc.js (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/DataTable/doc.js:333]:
```suggestion
    description: 'The icon indicating a column can be sorted. The icon will be replaced by the ascending / descending sort icons once clicked upon.',
```

Comment (ericsoderberghp) [reply src/js/components/DataTable/doc.js:333]: But, then I think we should also say something about it being restored when another column is sorted. We don't currently describe that level of detail for `dataTable.icons.ascending`, where we just say "The ascending icon.". Are you thinking we should document when the icons are used and then removed for all of them?

File: src/js/components/DataTable/stories/Custom.js (hunk 3 above)
Comment (ericsoderberghp) [reply src/js/components/DataTable/stories/Custom.js]: In the meantime, I've built a custom icon into the story.

File: src/js/components/DataTable/Header.js (hunk 1 above)
Comment (ShimiSun) [reply src/js/components/DataTable/Header.js:70]: Thank you for clarifying.

File: src/js/components/DataTable/doc.js (hunk 2 above)
Comment (ShimiSun) [reply src/js/components/DataTable/doc.js:333]: >But, then I think we should also say something about it being restored when another column is sorted

I agree. This level of detail might end up confusing.

File: src/js/components/DataTable/stories/Custom.js (hunk 3 above)
Comment (ShimiSun) [reply src/js/components/DataTable/stories/Custom.js]: And it looks great!

**Review by ShimiSun — APPROVED** [review]

**Pull request merged**